src/components/questions/QuestionsAnswers.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+class QuestionsAnswers extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      productId: null
+    }
+  }
+
+  componentDidMount() {
+    const { productId } = this.props;
+    this.setState({productId: productId})
```

Comment:
I believe that you would get the actual productId after getting null first from the App Parent Component. If this is the case, in my opinion, we can make a conditional statement here before storing into the state to prevent more rendering. 
As the below, only if the productId is not null, it will be stored in the state. 

If (productId) {
this.setState({
productId: productId
})
}

Reply:
I believe I actually ended up doing this before I even saw this comment, I think that should be okay now.